projects/observability/src/shared/components/explore-query-editor/order-by/explore-query-order-by-editor.component.ts

```diff
@@ -95,24 +95,30 @@ export class ExploreQueryOrderByEditorComponent implements OnChanges {
     }
   ];
   private readonly defaultDirection: GraphQlSortDirection = SortDirection.Asc;
-  private readonly incomingOrderByExpressionSubject: Subject<GraphQlOrderBy> = new ReplaySubject(1);
+  private readonly incomingOrderByExpressionSubject: Subject<ExploreOrderBy> = new ReplaySubject(1);
   private readonly contextSubject: Subject<TraceType> = new ReplaySubject(1);
 
   public constructor(private readonly metadataService: MetadataService) {
     this.selectedAggregation$ = combineLatest([this.contextSubject, this.incomingOrderByExpressionSubject]).pipe(
       switchMap(() => this.getCurrentSelectedAggregation())
     );
 
-    this.selectedMetric$ = combineLatest([this.contextSubject, this.incomingOrderByExpressionSubject]).pipe(
+    this.selectedAttribute$ = combineLatest([this.contextSubject, this.incomingOrderByExpressionSubject]).pipe(
       switchMap(() => this.getCurrentSelectedAttribute())
     );
 
-    this.selectedSortBy$ = this.incomingOrderByExpressionSubject.pipe(switchMap(() => this.getCurrentSelectedSortBy()));
+    this.selectedOrderBy$ = this.incomingOrderByExpressionSubject.pipe(
+      switchMap(() => this.getCurrentSelectedOrderBy())
+    );
 
-    this.metricOptions$ = this.contextSubject.pipe(switchMap(context => this.getAttributeOptionsForContext(context)));
+    this.attributeOptions$ = this.contextSubject.pipe(
+      switchMap(context => this.getAttributeOptionsForContext(context)),
+      shareReplay()
```

Comment:
Would this ever clean up? The shareReplay lives on after it's been unsubscribed and pretty sure we can drop it here. Plain share would probably accomplish what you're looking for, but honestly It's overkill for a cheap compute, it's cached upstream. (applies in multiple places in this file).

Reply:
Using share operator produces an incorrect result. Can I remove directly this shareReplay()?

Reply:
That may be the case if there's late subscribers, would need to go through the logic. In that case I'd just remove entirely.

Reply:
done. removed